common/rpc/interceptor/slow_request_logger.go

```diff
@@ -0,0 +1,74 @@
+package interceptor
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"go.temporal.io/server/common/dynamicconfig"
+	"go.temporal.io/server/common/log"
+	"go.temporal.io/server/common/log/tag"
+	"go.temporal.io/server/common/rpc/interceptor/logtags"
+	"go.temporal.io/server/common/tasktoken"
+	"google.golang.org/grpc"
+)
+
+// Certain types of methods are ignored as a rule.
+var ignoredMethodSubstrings = []string{"Poll", "GetWorkflowExecutionHistory"}
+
+type SlowRequestLoggerInterceptor struct {
+	logger       log.Logger
+	workflowTags *logtags.WorkflowTags
+	dc           *dynamicconfig.Collection
+}
+
+func NewSlowRequestLoggerInterceptor(
+	logger log.Logger,
+	dc *dynamicconfig.Collection,
+) *SlowRequestLoggerInterceptor {
+	return &SlowRequestLoggerInterceptor{
+		logger:       logger,
+		workflowTags: logtags.NewWorkflowTags(tasktoken.NewSerializer(), logger),
+		dc:           dc,
+	}
+}
+
+func (i *SlowRequestLoggerInterceptor) Intercept(
+	ctx context.Context,
+	request interface{},
+	info *grpc.UnaryServerInfo,
+	handler grpc.UnaryHandler,
+) (interface{}, error) {
+	threshold := dynamicconfig.SlowRequestLoggingThreshold.Get(i.dc)()
```

Comment:
Ideally we would do the exclusion before the call to get the dynamic config, to avoid that call if we're excluding anyway